bentoml/_internal/runner/runnable.py

```diff
@@ -53,48 +36,54 @@ def supports_nvidia_gpu(self) -> bool:
     def supports_multi_threading(self) -> bool:
         ...
 
-    @overload
-    @staticmethod
-    def method(
-        batchable_or_method: WrappedMethod,
-        batch_dim: BatchDimType = 0,
-        input_spec: LazyType[t.Any] | t.Tuple[LazyType[t.Any], ...] | None = None,
-        output_spec: LazyType[t.Any] | None = None,
-    ) -> WrappedMethod:
-        ...
-
-    @overload
-    @staticmethod
-    def method(
-        batchable_or_method: bool,
+    @classmethod
+    def add_method(
+        cls,
+        method: t.Callable[..., t.Any],
+        name: str,
+        *,
+        batchable: bool = False,
         batch_dim: BatchDimType = 0,
         input_spec: LazyType[t.Any] | t.Tuple[LazyType[t.Any], ...] | None = None,
         output_spec: LazyType[t.Any] | None = None,
-    ) -> t.Callable[[WrappedMethod], WrappedMethod]:
-        ...
+    ):
+        setattr(
+            cls,
+            name,
+            Runnable.method(
+                method,
+                batchable=batchable,
+                batch_dim=batch_dim,
+                input_spec=input_spec,
+                output_spec=output_spec,
+            ),
+        )
 
     @staticmethod
     def method(
-        batchable_or_method: bool | WrappedMethod = False,  # type: ignore (pyright bug?)
+        meth: WrappedMethod | None = None,
```

Comment:
user who's using `Runnable.method` as a decorator may get confused by this parameter?  I will try to refactor the to_method to get rid of this parameter here

previously with the `batchable_or_method`, it was trying to support using the decorator without any parameters, so both of the following works:

```
@Runnable.method
def __call__():
   ...

@Runnable.method()
def __call__():
   ...
```